### PR TITLE
Fixes #16507 Subdags are now created with pause status based on their parent.

### LINKED ISSUE #16507: Adding SubDag to an already unpaused dag created in paused state
**Apache Airflow version**: Checked in 2.0.2 as well as 2.2

**What happened**:

Adding a new subdag to an already unpaused dag creates the subdag task in paused state. This causes the subdag tasks to not start.
 
image
</img>

**What you expected to happen**:

I expect the subdag to follow the same state of the parent dag. 
PS: Turning the parent dag off and turning it back on fixes the state subdag.

**How to reproduce it**:
```
def simple_subdag(parent_dag_name, subdag_id):

    with DAG(
    dag_id=f'{parent_dag_name}.{subdag_id}',
    default_args=args,
    start_date=datetime(2021, 6, 9),
    schedule_interval="@once",
    ) as subdag:
        start_subdag = DummyOperator(
            task_id='start_subdag',
        )

        start_subdag

    return subdag
```

```
with DAG(
    dag_id='test_subdag',
    default_args=args,
    start_date=datetime(2021, 6, 9),
    schedule_interval="@once",
) as dag:

    run_this_first = DummyOperator(
        task_id='run_this_first',
    )

    do_something = SubDagOperator(
        task_id='do_something',
        subdag=simple_subdag('test_subdag','do_something')
    )
    run_this_first >> do_something

```
1. Create a dag
2. turn the dag on
3. add a new subdag task and retrigger the dag.

**Willing to submit PR**:
Yes

### PR DESCRIPTION
**Pull request by ManiBharataraju**
closes: #16507
In this PR I'm checking the pause status of the parent and assigning it to the subdag if they are newly added.

### CODE REVIEW

File: airflow/models/dag.py
Comment (uranusjr) [inline airflow/models/dag.py:2405]: This feels a bit hacky to me. Can the subdag not be created with `is_paused_upon_creation` correctly set to True? Why not?

Comment (ManiBharataraju) [reply airflow/models/dag.py:2405]: @uranusjr Per my understanding, the `is_subdag` parameter is set to true only in the `_bag_dag` method, and setting `is_paused_upon_creation` there will impact existing subdags as well(don't think that will negatively impact but still). Changing is_paused there will be overwritten by this method for newly added ones. So, I chose to make the changes here. Please correct me if my understanding is wrong.
https://github.com/apache/airflow/blob/28e285ef9a4702b3babf6ed3c094af07c017581f/airflow/models/dagbag.py#L447-L450

Comment (ManiBharataraju) [reply airflow/models/dag.py:2405]: @uranusjr - any thoughts?

Comment (uranusjr) [reply airflow/models/dag.py:2405]: > setting is_paused_upon_creation there will impact existing subdags as well (don't think that will negatively impact but still).

I’m definitely missing something, because I thought this is what this PR is for? When a parent DAG is set to pause, its subdags should also be paused?

Comment (ManiBharataraju) [reply airflow/models/dag.py:2405]: @uranusjr - Okay. So what is happening is, let's say you create a dag and deploy it and it is running. At a later point, you want to add an enhancement to the dag which is a subdag. You would expect the subdag to run in the next schedule. However, what is happening is, the newly added subdag is created in off state because `is_paused_upon_creation` by default is True. My PR fixes this issue by checking the status of the parent dag for such cases only. 

The workaround for this currently is, turn off the parent dag and turn it back on so that both the parent and subdag is in the same state.

> When a parent DAG is set to pause, its subdags should also be paused?

This is currently working fine and this PR does not change anything wrt this.
If you still have questions, I can ping you over the slack channel. Thanks!

Comment (uranusjr) [reply airflow/models/dag.py:2405]: I’m still not following so please bear with me…

> However, what is happening is, the newly added subdag is created in off state because is_paused_upon_creation by default is False.

If the default is False (i.e. not paused), why is the subdag created in the off state?

----

The default `dags_are_paused_at_creation` value is actually True, so by default, dags are created in in the paused state. So when the parent DAG is unpaused, the newly-created subdag would indeed be in a different state. Is this what you’re trying to say? (but not describing the true/false on/off values correctly… which is understandable, the various negative value names are difficult to keep straight)

----

Assuming the above is the case, setting a subdag to unpaused by default if the parent dag is unpaused does sound like a reasonable thing to do, and the patch is also right (from what I can tell). But that would arguably be a backward incompatibility behavioural change I’m not sure we should just do. I hope another maintainer can provide second opinion on this.

Comment (ManiBharataraju) [reply airflow/models/dag.py:2405]: > If the default is False (i.e. not paused), why is the subdag created in the off state?

My bad, Sorry. I just fixed my comment. It is true by default.

> The default dags_are_paused_at_creation value is actually True, so by default, dags are created in in the paused state. So when the parent DAG is unpaused, the newly-created subdag would indeed be in a different state. Is this what you’re trying to say? (but not describing the true/false on/off values correctly… which is understandable, the various negative value names are difficult to keep straight)

Yes right.

**Review by uranusjr — APPROVED** [review]
(To another maintainer) This is correct from what I can tell, but there are implications so please read my comment https://github.com/apache/airflow/pull/16508/files#r670205197 before you decide to merge this.

**Comment by potiuk** [comment]
Can you plese rebase this one @ManiBharataraju ?

**Comment by ManiBharataraju** [comment]
@potiuk - Apologies. Completely missed this. Should I just do a git pull --rebase on my branch? Could you help.

**Comment by potiuk** [comment]
see the [docs](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#id14)

**Pull request closed without merging**
